package.json

```diff
@@ -36,27 +36,116 @@
+          "enum": [
+            "discover",
+            "cabal-fmt",
+            "custom"
+          ],
+          "enumItemLabels": [
+            "Discover",
+            "cabal-fmt",
+            "Custom"
+          ],
+          "markdownEnumDescriptions": [
+            "Automatically determines the appropriate command.",
+            "Uses `cabal-fmt`.",
+            "Uses the command in `#purple-yolk.cabal.formatter.command#`."
+          ],
+          "type": "string"
+        },
         "purple-yolk.haskell.formatter.command": {
-          "default": "ormolu --no-cabal",
+          "default": "",
           "description": "The command to run when formatting Haskell files. This should accept an unformatted Haskell file as input on STDIN. It should emit a formatted Haskell file to STDOUT. STDERR will be ignored.",
           "type": "string"
         },
+        "purple-yolk.haskell.formatter.mode": {
+          "default": "discover",
+          "description": "The command to format a Haskell file.",
+          "enum": [
+            "discover",
+            "fourmolu",
+            "ormolu",
+            "custom"
+          ],
+          "enumItemLabels": [
+            "Discover",
+            "Fourmolu",
+            "Ormolu",
+            "Custom"
+          ],
+          "markdownEnumDescriptions": [
+            "Automatically determines the appropriate command.",
+            "Uses `fourmolu`.",
+            "Uses `ormolu`.",
+            "Uses the command in `#purple-yolk.haskell.formatter.command#`."
+          ],
+          "type": "string"
+        },
         "purple-yolk.haskell.interpreter.command": {
-          "default": "cabal repl --repl-options -ddump-json",
-          "description": "The command to launch a Haskell interpreter. This should be something like `cabal repl` or `stack ghci`. It should also pass `-ddump-json` to GHCi.",
+          "default": "",
+          "markdownDescription": "This should pass `-ddump-json` to GHC.",
+          "type": "string"
+        },
+        "purple-yolk.haskell.interpreter.mode": {
+          "default": "discover",
+          "description": "The command to launch a Haskell interpreter.",
+          "enum": [
+            "discover",
+            "cabal",
+            "stack",
+            "ghci",
+            "custom"
+          ],
+          "enumItemLabels": [
+            "Discover",
+            "Cabal",
+            "Stack",
+            "GHCi",
+            "Custom"
+          ],
+          "markdownEnumDescriptions": [
+            "Automatically determines the appropriate command.",
+            "Uses `cabal repl`.",
+            "Uses `stack ghci`.",
+            "Uses `ghci`. Only works with a single file.",
+            "Uses the command in `#purple-yolk.haskell.interpreter.custom#`."
+          ],
           "type": "string"
         },
         "purple-yolk.haskell.linter.command": {
-          "default": "hlint --json --no-exit-code -",
+          "default": "",
```

Comment:
This is a noteworthy change. If you previously customized the HLint command, you'll now have to set the mode to "custom" for that command to be used.